### PR TITLE
Add mistral actions to the release process

### PR DESCRIPTION
**Pull request by m4dcoder**

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1449795414012071

File: actions/mistral_update_puppet.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -e
+
+VERSION=$1
+REPO_PUPPET=$2
+REPO_PUPPET_BRANCH=$3
+
+GIT=`which git`
+
+MISTRAL_BRANCH=st2-${VERSION}
```
Comment (manasdk) [inline actions/mistral_update_puppet.sh]: Those lines should likely go after the version format check

File: actions/workflows/st2_prepare_for_release.yaml [hunk 2]
```diff
@@ -103,6 +103,12 @@
         hosts: "{{get_build_server.result[0]}}"
         repo: "{{st2_express_repo_target}}"
         branch: "v{{st2_release_base_ver}}"
+      on-success: "setup_mistral_for_release"
+    -
+      name: "setup_mistral_for_release"
+      ref: "st2cd.mistral_release_candidate"
```
Comment (manasdk) [inline actions/workflows/st2_prepare_for_release.yaml:109]: Existing action?

**Comment by manasdk** [comment]
overall looks reasonable

File: actions/workflows/st2_prepare_for_release.yaml (hunk 2 above)
Comment (m4dcoder) [inline actions/workflows/st2_prepare_for_release.yaml:109]: Yes, this exists in the st2cd packs today.  It is what I run to release candidate for the last few releases.

File: actions/mistral_update_puppet.sh (hunk 1 above)
Comment (m4dcoder) [inline actions/mistral_update_puppet.sh]: Got it.

**Pull request merged**